### PR TITLE
[Birthday] Output special message if bot's birthday is set

### PR DESCRIPTION
**Pull request by Injabie3**
### Description of the changes
This PR will do a few things:
1. Add an interlock to wait for the bot to be ready before starting the loop, or else `bot.user` is `None` when the bot is starting up.
2. Add a helper `getBirthdayMessage` that gets a random message from `CANNED_MESSAGES` if it's not the bot's birthday, else returns `BOT_BIRTHDAY`. This helper replaces the background loop's version of fetching the message in order to output a special message if the bot's birthday has been configured.
4. Move config, logger, and background init to helpers in order to avoid running into initialization issues while unit testing helper functions.
5. Add a unit test for the `getBirthdayMessage` helper.

Screenshot from 2022-10-03 23-59-59

### CODE REVIEW

File: cogs/birthday/birthday.py [hunk 1]
```diff
@@ -41,10 +39,22 @@ def __init__(self, bot: Red):
...
         # On cog load, we want the loop to run once.
         self.lastChecked = datetime.now() - timedelta(days=1)
         self.bgTask = self.bot.loop.create_task(self.birthdayLoop())
 
+    # Class constructor
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.bgTask: asyncio.Task = None
+        self.config = None
+        self.lastChecked = None
```
Comment (quachtridat) [inline cogs/birthday/birthday.py]: Might as well add type hints for these vars.
```suggestion
        self.config: Config = None
        self.lastChecked: datetime = None
```

File: cogs/birthday/birthday.py [hunk 2]
```diff
@@ -716,7 +746,8 @@ async def _dailyAdd(self):  # pylint: disable=too-many-branches
                                 continue
                             try:
                                 msg = choice(CANNED_MESSAGES)
```
Comment (quachtridat) [inline cogs/birthday/birthday.py]: Can have this removed.
```suggestion
```

File: cogs/birthday/constants.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
     KEY_ALLOW_SELF_BDAY: False,
 }
 
+BOT_BIRTHDAY = "Wow! It's my own birthday! Happy birthday to myself!"
```
Comment (quachtridat) [inline cogs/birthday/constants.py]: Perhaps a better name could be used? I mean, it sounds more like this constant is about the bot's birthday than the message sent on the day.
```suggestion
BOT_BIRTHDAY_MSG = "Wow! It's my own birthday! Happy birthday to myself!"
```

File: cogs/birthday/birthday.py [hunk 4]
```diff
@@ -582,13 +592,33 @@ async def toggleSelfBirthday(self, ctx: Context):
...
+        member: discord.Member
+            The member that we want the birthday message for.
+
+        Returns
+        -------
+        str
+            The birthday message, already formatted.
+        """
+        if self.bot.user.id == member.id:
+            return BOT_BIRTHDAY
```
Comment (quachtridat) [inline cogs/birthday/birthday.py]: Perhaps a better name could be used? I mean, it sounds more like this constant is about the bot's birthday than the message sent on the day.
```suggestion
            return BOT_BIRTHDAY_MSG
```

File: cogs/birthday/testBirthday.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+import discord
+import pytest
+
+
+from .birthday import Birthday
+from .constants import CANNED_MESSAGES
+
+BOT_BIRTHDAY = "Wow! It's my own birthday! Happy birthday to myself!"
```
Comment (quachtridat) [inline cogs/birthday/testBirthday.py]: Why is this not imported form `.constants`?

File: cogs/birthday/testBirthday.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    def testGetBirthdayMessageWhenUserIsBot(self, mocker):
+        # Do not initialize config, logger, and background task
+        mocker.patch("birthday.Birthday.initializeConfigAndLogger")
+        mocker.patch("birthday.Birthday.initializeBgTask")
+
+        birthday = Birthday(self.bot)
+        mockUser = MockUser(1234, "Ren")
+
+        msg = birthday.getBirthdayMessage(mockUser)
+        assert msg == BOT_BIRTHDAY
```
Comment (quachtridat) [inline cogs/birthday/testBirthday.py]: Perhaps a better name could be used? I mean, it sounds more like this constant is about the bot's birthday than the message sent on the day.
```suggestion
        assert msg == BOT_BIRTHDAY_MSG
```

**Review by quachtridat — CHANGES_REQUESTED** [review]
Overall fine. Few points below. Would be nice if you can include an image of manual testing done.

File: cogs/birthday/constants.py (hunk 3 above)
Comment (Injabie3) [reply cogs/birthday/constants.py]: ack

File: cogs/birthday/birthday.py (hunk 2 above)
Comment (Injabie3) [reply cogs/birthday/birthday.py]: Oops, removed now.

**Comment by Injabie3** [comment]
> Overall fine. Few points below. Would be nice if you can include an image of manual testing done.

Updated

**Review by quachtridat — APPROVED** [review]
🚢 It!

**Pull request merged**